runtime/lua/vim/lsp/util.lua

```diff
@@ -481,6 +481,29 @@ function M.jump_to_location(location)
   return true
 end
 
+--- Preview a location in a floating windows
+---
+--- behavior depends on type of location:
+---   - for Location, range is shown (e.g., function definition)
+---   - for LocationLink, targetRange is shown (e.g., body of function definition)
+---   - for an array of either, the first element is shown
+---
+--@param location Location, LocationLink, or array thereof
+--@return bufnr,winnr buffer and window number of floating window or nil
+function M.preview_location(location)
```

Comment:
Does this actually handle a list of locations? It doesn't look like it from reading the source.

Reply:
It handles a list in the sense that it doesn't run away screaming -- if there are multiple locations, it just shows the first one. I don't know any more reasonable approach here -- do you have a suggestion?

Reply:
oh, no, you're right! I moved that into the callback. Would this be reasonable here? (I'm thinking "no"?)

Reply:
I think it's fine to handle them here. LSP spec has this all over the place (which is annoying), where it passes `X or X[]`, so it goes more w/ the spirit of LSP IMO. 

Additionally, if someone wants the second one in the list... they can just parse the list and use only the second one. So I'd say handling a list would be helpful and in the spirit of spec.

Reply:
In my original implementation, that was (and still is) the job of the callback: parse the response, extract the (first) location, and preview it using that utility function. In particular, it mirrors the `jump_to_location` function, which also only takes a single location.

Reply:
Nvm, chatted on gitter and it seems fine to just update docs to not take a list.